app/scripts/streams.js

```diff
@@ -0,0 +1,121 @@
+/**
+ *  RxJS in action
+ *  Chapter #
+ *  @author Paul Daniels
+ *  @author Luis Atencio
+ */
+
+const {createStore} = Redux;
+const {ofType, createDispatcher} = DispatcherUtils;
+
+const dispatcher = createDispatcher(createStore(bankingApp, {
+  accounts: {checking: 100, savings: 100},
+  transactions: {transactions: [], limit: 10, skip: 0},
+  messages: [],
+  searches: {query: '', results: []}
+}));
+
+const balanceActions = {
+  withdraw: () => dispatcher.dispatch({type: PROCESS_TRANSACTION, factor: -1}),
+  deposit: () => dispatcher.dispatch({type: PROCESS_TRANSACTION, factor: 1}),
+  amount: (value) => dispatcher.dispatch({type: UPDATE_AMOUNT, amount: value}),
+  account: (value = 'checking') => dispatcher.dispatch({type: UPDATE_ACCOUNT, account: value})
+};
+
+dispatcher.include(handleSearch);
+dispatcher.include(handleMessages);
+dispatcher.include(processUserTransaction);
+dispatcher.include(processInterest);
+dispatcher.include(handleTransaction);
+
+function handleMessages(actions$, store$) {
+  return Rx.Observable.interval(250)
+    .withLatestFrom(store$.pluck('messages'))
+    .filter(([_, messages]) => messages.length > 0)
+    .mapTo(degradeMessages());
+}
```

Comment:
This is perhaps a little contrived, but it was a test case to force messages to degrade over a period of time before disappearing.